nifi-commons/nifi-flow-encryptor/src/test/java/org/apache/nifi/flow/encryptor/StandardFlowEncryptorTest.java

```diff
@@ -89,7 +100,145 @@ public void testProcessNoEncrypted() {
         assertEquals(property, outputProperty);
     }
 
+    @Test
+    public void testProcessJson() throws JsonProcessingException {
+        final String password = "This is a bad password";
+        final String encryptedPassword = String.format(ENCRYPTED_FORMAT, inputEncryptor.encrypt(password));
+
+        final String sampleFlowJson = getSampleFlowJson(encryptedPassword);
+
+        final InputStream inputStream = new ByteArrayInputStream(sampleFlowJson.getBytes(StandardCharsets.UTF_8));
+        final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        flowEncryptor.processFlow(inputStream, outputStream, inputEncryptor, outputEncryptor);
+
+        final ObjectMapper mapper = new ObjectMapper();
+
+        final String outputFlowJson = new String(outputStream.toByteArray());
+
+        assertNotEquals(mapper.readTree(sampleFlowJson), mapper.readTree(outputFlowJson));
+
+        final JsonNode outputJsonNode = mapper.readTree(outputFlowJson);
+
+        final String outputEncryptedPassword = outputJsonNode
+                .path("processors")
+                .get(0)
+                .path("properties")
+                .path("password")
+                .asText();
+
+        final Matcher outputMatcher = PATTERN.matcher(outputEncryptedPassword);
+        assertTrue(outputMatcher.matches());
+        assertEquals(password, outputEncryptor.decrypt(outputMatcher.group(1)));
+    }
+
+    @Test
+    public void testProcessXml() {
+        final String password = "This is a bad password!";
+        final String encryptedPassword = String.format(ENCRYPTED_FORMAT, inputEncryptor.encrypt(password));
+        final String sampleFlowXml = getSampleFlowXml(encryptedPassword);
+        final InputStream inputStream = new ByteArrayInputStream(sampleFlowXml.getBytes(StandardCharsets.UTF_8));
+        final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        flowEncryptor.processFlow(inputStream, outputStream, inputEncryptor, outputEncryptor);
+
+        final String outputXml = new String(outputStream.toByteArray());
+
+        final Source control = Input.fromString(sampleFlowXml).build();
+        final Source test = Input.fromString(outputXml).build();
+        DifferenceEngine diff = new DOMDifferenceEngine();
+        diff.addDifferenceListener((comparison, outcome) -> {
+            final Matcher inputMatcher = PATTERN.matcher(comparison.getControlDetails().getValue().toString());
+            final Matcher outputMatcher = PATTERN.matcher(comparison.getTestDetails().getValue().toString());
+            if (inputMatcher.matches() && outputMatcher.matches()) {
+                assertEquals(inputEncryptor.decrypt(inputMatcher.group(1)), outputEncryptor.decrypt(outputMatcher.group(1)));
+            }
+        });
+        diff.compare(control, test);
```

Comment:
Introducing the XMLUnit dependency seems like a lot just to perform this comparison. It seems better to avoid the additional dependencies and stay with a simpler comparison.